src/core/raps/execute.ts

```diff
@@ -177,11 +182,14 @@ export const walletExecuteRap = async (
         };
         const { hash } = await executeAction(actionParams);
         hash && (await waitForNodeAck(hash, wallet.provider));
+        if (index === actions.length - 1) {
```

Comment:
What's the logic for this check?

Reply:
it returns tx hash of the final action in the array. Realizing now that I needed this in my original implementation but don't anymore. I can remove returning this if you want, but I wouldn't be surprised if having the tx hash of the last action in a rap ends up being useful.